model/coverage.py

```diff
@@ -228,10 +228,18 @@ def update(self, start=None, finish=None, achievements=None,
         """Use a single method to update all the fields that aren't
         used to identify a Timestamp.
         """
-        self.start = start
-        self.finish = finish
-        self.achievements = achievements
-        self.counter = counter
+
+        if start is not None:
```

Comment:
What happens if any of these values is `None`?

Reply:
The only one I can think of that would be `None` normally is `exception`. But if you really wanted to set `achievements` to None, you would have to do it by manipulating the `Timestamp` object directly. You can no longer do it by calling `update()`.

Another possibility is to introduce a constant `NO_VALUE` and make that the default. That way it'll be clear whether a value is missing because the caller wants it to be set to `None`, or whether it's missing because the caller didn't specify a value.